articles/112_generated_interfaces_cpp.md

```diff
@@ -28,16 +30,25 @@ This article specifies the generated C++ code for ROS interface types defined in
 ## Namespacing
 
 All code of a ROS package should be defined in a namespace named after the package.
-To separate the generated code from other code within the package it is defined in a sub namespace:
+To separate the generated code from other code within the package it is defined in a sub-namespace:
+
 * namespace for ROS messages: `package_name::msg`.
 * namespace for ROS services: `package_name::srv`.
 
+<div class="alert alert-warning" markdown="1">
+  **Note:** Those namespaces are not consistent with ROS 1 usage.
+</div>
+
 
 ## Generated files
 
 Following the C++ style guide of ROS 2 the namespace hierarchy is mapped to a folder structure.
 The filenames use lowercase alphanumeric characters with underscores for separating words and end with either `.hpp` or `.cpp`.
 
+<div class="alert alert-warning" markdown="1">
+  **Note:** The suggested naming is not consistent with ROS 1 usage.
+</div>
+
```

Comment:
I believe the header file names should be consistent with ROS 1:
```C++
#include <my_msgs/MyMessageName.h>
```
The differences are: there is no intermediate directory name; the message name is CamelCase, like the message definition; and the header suffix is `.h` as recommended by the [ROS C++ Style Guide](http://wiki.ros.org/CppStyleGuide#Files).

The argument for changing it is to make the static ROS 1 bridge easier to write (and possibly other, similar hybrid packages). I don't believe that justifies the cost of updating every message header reference in every existing ROS C++ package.